### DISCUSSION TITLE
Refresh tokenを用いたAccess tokenの更新に対応しませんか

### DISCUSSION THREAD

**Discussion by ixsiid** [tateisu/SubwayTooter#256] [Ideas]
Mastodonが公式には、永続的なAccess tokenを求めてることは知っています。

しかし、標準的なOAuthはRefresh tokenを用いた定期的な更新を推奨しており、クラウドサービスプロバイダーもそれを提供しています。
永続的なAccess tokenを提供するため、インスタンス運営者はクラウドサービスのIaaSを使えないことがあります。

Mastodon公式がこれに否定的なことは知っていますが、本来OAuthのフローはMastodonとは無関係であり
Third partyアプリから対応することも有用に思います。

**Comment by tateisu** [comment]
アクセストークンはMastodonではプッシュ購読を保存するスロットの役割を果たしています。
リフレッシュトークンを使う事がまず可能なのか、既存のプッシュ購読がどうなるのか、またWebUIのユーザ設定の認証済みアプリ一覧が期待通りに動くのか、などサーバ側挙動がどうなのかをまず教えてください。

次に、プッシュ配送に使われるSTアプリサーバはプッシュ購読を識別するのにアクセトークンのSHA256ダイジェストを使っています。アクセストークンが変化する場合、それぞれのプッシュ購読スロットはどのように区別されるべきでしょうか。

最後に、STはバックアップやエクスポートでアクセストークンを保存し、それらはしばしば別の端末で復元されます。この挙動についての評価は色々ありますが、ユーザからの概ね好評です。リフレッシュトークン移行はこの機能に影響するかもしれません。

2023年9月18日(月) 20:35 Tal ***@***.***>:

**Comment by ixsiid** [comment]
Mastodonのプッシュ購読に影響を及ぼすこと、またその仕様について調査ができていませんでした。
どのような影響がでるかわかりましたら、またご連絡します。
